controllers/openstackdataplane_controller.go

```diff
@@ -96,20 +96,20 @@ func (r *OpenStackDataPlaneReconciler) Reconcile(ctx context.Context, req ctrl.R
 
 	// Always patch the instance status when exiting this function so we can persist any changes.
 	defer func() {
-		// update the overall status condition if service is ready
-		if instance.IsReady() {
```

Comment:
IsReady is now used to check when a CR depends on another CR and we want to ensure the dependent CR is ready before continue reconcile (eg Heat -> KeystoneAPI, OpenStackControlPlane -> Heat). This might not needed in dataplane-operator unless we want to create OpenStackDataplane CRs via meta-operator, but I tend to leave it for consistency.

I'd avoid using `AllSubConditionsIsTrue` to IsReady. The `Ready` condition is meant to present readiness of the CR, and controller is responsible to update it. Instead we should use ReadyCondition, instead of DeploymentReadyCondition, in IsReady, I think

Reply:
I think that is the conversation I'm recalling too. About the status conditions being set in the controller, and the `IsReady()` used as an interface. I remember there were a few iterations on this in the other operators before arriving at current state.

Thanks for the clarification @kajinamit ++

Reply:
ok, that works. IsReady and AllSubConditionIsTrue both equate to the CR being ready.